Server/routes.go

```diff
@@ -1151,18 +1153,22 @@ func (r *Routes) CreateTSNE(w http.ResponseWriter, req *http.Request) {
 				return
 			}
 
-			// make it nonblocking...
-			go func() {
-				err = collection.CreateTSNE(tsne.Dimensions, tsne.Iterations, tsne.Learningrate)
-				if err != nil {
-					Logger.Log.Log("Error creating TSNE: " + err.Error())
-					return
-				}
-			}()
-			w.WriteHeader(http.StatusOK)
+			err = collection.CreateTSNE(tsne.Dimensions, tsne.Iterations, tsne.Learningrate)
+			if err != nil {
+				Logger.Log.Log("Error creating TSNE: " + err.Error())
+				return
+			} else {
```

Comment:
do you think waitingtimes are not a thing here?

Reply:
If we don't block the consequent call to gettsnedata will fail... How can we address this?